compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1639,16 +1643,14 @@ public void visit(BLangMapLiteral astMapLiteralExpr) {
 
         BType type = astMapLiteralExpr.getBType();
         this.env.isInArrayOrStructure++;
-        visitTypedesc(astMapLiteralExpr.pos, type, Collections.emptyList(), getAnnotations(type.tsymbol, this.env));
 
-        BIRVariableDcl tempVarDcl =
-                new BIRVariableDcl(astMapLiteralExpr.getBType(), this.env.nextLocalVarId(names),
-                                   VarScope.FUNCTION, VarKind.TEMP);
+        BIRVariableDcl tempVarDcl = new BIRVariableDcl(type, this.env.nextLocalVarId(names), VarScope.FUNCTION,
+                                                       VarKind.TEMP);
         this.env.enclFunc.localVars.add(tempVarDcl);
         BIROperand toVarRef = new BIROperand(tempVarDcl);
-
-        setScopeAndEmit(new BIRNonTerminator.NewStructure(astMapLiteralExpr.pos, toVarRef, this.env.targetOperand,
-                                               generateMappingConstructorEntries(astMapLiteralExpr.fields)));
+        BIROperand typeDesc = new BIROperand(getOrCreateTypedescVariable(type, astMapLiteralExpr.pos));
```

Comment:
Dont we need to bump the BIRVERSION because of this? @rdulmina @warunalakshitha

Reply:
Yes, will bump